### PR TITLE
refactor: Import createFormatFunction + associated from UMA

### PR DESCRIPTION
**Pull request by pxrl**
This unfortunately introduces direct Web3.js dependencies, but we need to import it as-is before we can clean it up.

[update]: No issue anymore - this is a very simple change now.

### CODE REVIEW

File: src/utils/common.ts
Comment (pxrl) [reply src/utils/common.ts]: Explanation: there are actually two variants of `ConvertDecimals()` - one supporting `Web3.js BN` and the other supporting `ethers BigNumber`. It's not strictly required because raw number and string types are used on both input and output sides, but there's a constraint in that `FormattingUtilsInternal.ts` is not intended to import from other files, and it ideally shouldn't depend directly on `ethers BigNumber` either.

So I propose that we stick with the Web3 variant hidden in `FormattingUtilsInternal.js` and clean this up post-import.

**Review by nicholaspai — APPROVED** [review]

File: src/utils/common.ts
Comment (james-a-morris) [inline src/utils/common.ts:82]: Could we return this in terms of `BigNumberish` or directly as a `BigNumber`?

File: src/utils/FormattingUtilsInternal.ts
Comment (james-a-morris) [inline src/utils/FormattingUtilsInternal.ts:1]: Do we have a Linear task for this? I like the intermediate step

**Review by james-a-morris — APPROVED** [review]
LGTM pending the creation of a linear task for making sure we remove `FormattingUtilsInternal.ts`

File: src/utils/FormattingUtilsInternal.ts
Comment (pxrl) [reply src/utils/FormattingUtilsInternal.ts:1]: Shouldn't be necessary anymore - I was able to re-work this after your tsdx migration and it's all much simpler now.

File: src/utils/common.ts
Comment (pxrl) [reply src/utils/common.ts]: I was able to drop this import after merging in your recent changes :+1:

File: src/utils/common.ts
Comment (pxrl) [reply src/utils/common.ts]: Update - it was possible to drop the duplicate imports after merging in the latest changes to master.

**Pull request merged**
